### PR TITLE
Remove dependency on GCS for AP_AccelCal and AP_InertialSensor for AP_Periph build

### PR DESCRIPTION
**Pull request by bugobliterator**
The following changes should allow building InertialSensor library that used by some compass drivers, example peripheral CubeOrange-periph. And also opens the door for follow up PR to support transmitting INS data as well.

### CODE REVIEW

**Review by magicrub — APPROVED** [review]
The changes look ok, but I don't see anything taking advantage of the two new libraries being pulled in.

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 1]
```diff
@@ -102,15 +109,17 @@ void AP_AccelCal::update()
                                 fail();
                                 return;
                         }
-                        _printf("Place vehicle %s and press any key.", msg);
+                        _printf("Place vehicle %s and press any key.", msg);
```
Comment (amilcarlucas) [inline libraries/AP_AccelCal/AP_AccelCal.cpp:112]: pure whitespace change

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 2]
```diff
@@ -168,10 +177,18 @@ void AP_AccelCal::update()
             _last_position_request_ms = now;
             switch (_last_result) {
                 case ACCEL_CAL_SUCCESS:
+#if !defined(HAL_BUILD_AP_PERIPH)
                     _gcs->send_accelcal_vehicle_position(ACCELCAL_VEHICLE_POS_SUCCESS);
-                    break;
+#else
+                    _printf("AccelCal Successful");
+#endif
+                break;
```
Comment (amilcarlucas) [inline libraries/AP_AccelCal/AP_AccelCal.cpp:185]: break indentation

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 3]
```diff
@@ -79,6 +85,7 @@ void AP_AccelCal::update()
 
                     if(_use_gcs_snoop) {
                         const char *msg;
+                        (void)msg; // just to silence compiler
```
[the remark this answers is not in the document]
Comment (bugobliterator) [reply libraries/AP_AccelCal/AP_AccelCal.cpp:88]: we are not loosing _printf on all AP_Periph builds, we have can_printf on most boards that have enough flash.

**Comment by tridge** [comment]
delay till peters AP_Logging for periph is merged

File: libraries/AP_AccelCal/AP_AccelCal.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (peterbarker) [reply libraries/AP_AccelCal/AP_AccelCal.cpp:88]: See https://github.com/ArduPilot/ardupilot/pull/17842 for an alternative to this.

The new `_printf` define in this file would get `while (0) {(void)(format); UNUSED_VARARGS(args); }`

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 4]
```diff
@@ -110,7 +117,9 @@ void AP_AccelCal::update()
                 uint32_t now = AP_HAL::millis();
                 if (now - _last_position_request_ms > AP_ACCELCAL_POSITION_REQUEST_INTERVAL_MS) {
                     _last_position_request_ms = now;
+#if !defined(HAL_BUILD_AP_PERIPH)
```
Comment (peterbarker) [inline libraries/AP_AccelCal/AP_AccelCal.cpp:120]:
```suggestion
`#ifndef HAL_NO_GCS  || !HAL_NO_GCS`
```

I think that's the pattern required.

It should be
```suggestion
#if !HAL_NO_GCS
```

but I'm not sure it is always defined ATM.  And it really should be `HAL_GCS_ENABLED`

In any case, we should trust the define.

Similarly elsewhere in here.

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 5]
```diff
@@ -168,10 +177,18 @@ void AP_AccelCal::update()
             _last_position_request_ms = now;
             switch (_last_result) {
                 case ACCEL_CAL_SUCCESS:
+#if !defined(HAL_BUILD_AP_PERIPH)
```
Comment (peterbarker) [inline libraries/AP_AccelCal/AP_AccelCal.cpp:180]: breaking out a `send_vehicle_position` would reduce the number of places we need to `#if` - readability win

File: libraries/AP_CANManager/AP_CANTester.h [hunk 6]
```diff
@@ -18,7 +18,7 @@
 #include "AP_CANDriver.h"
 #include <AP_HAL/Semaphores.h>
 #include <AP_UAVCAN/AP_UAVCAN.h>
-#if HAL_MAX_CAN_PROTOCOL_DRIVERS > 1 && !HAL_MINIMIZE_FEATURES && HAL_ENABLE_LIBUAVCAN_DRIVERS
+#if HAL_MAX_CAN_PROTOCOL_DRIVERS > 1 && !HAL_MINIMIZE_FEATURES && HAL_ENABLE_CANMANAGER
```
Comment (peterbarker) [inline libraries/AP_CANManager/AP_CANTester.h:21]: Our pattern is
`HAL_CANMANAGER_ENABLED`

May as well conform if we're changing it everywhere

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 7]
```diff
@@ -922,9 +922,11 @@ AP_InertialSensor::detect_backends(void)
 #endif
 
     uint8_t probe_count = 0;
+    (void)probe_count;
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp:925]: Alternatively we could have `HAL_INS_ENABLED`.  If you want the INS lib in AP_Periph you have to override that - and if you subsequently don't allow backends you'd die.  But at least you'd have an idea of what's going on...

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_AccelCal/AP_AccelCal.cpp (hunk 5 above)
Comment (bugobliterator) [reply libraries/AP_AccelCal/AP_AccelCal.cpp]: It's only three places, also the behaviour is similar in only these two cases, for the third one it's supposed to just ignore it, don't think its worth it to create another thing to scroll to 😄 .

File: libraries/AP_AccelCal/AP_AccelCal.h [hunk 8]
```diff
@@ -40,7 +40,9 @@ class AP_AccelCal {
     bool running(void) const;
 
 private:
+#if !defined(HAL_BUILD_AP_PERIPH)
```
Comment (peterbarker) [inline libraries/AP_AccelCal/AP_AccelCal.h:43]:
```suggestion
#ifndef HAL_NO_GCS
```

Missed one.

**Review by peterbarker — CHANGES_REQUESTED** [review]
This does look like it should've been two PRs, one for the `HAL_CANMANAGER_ENABLED` change.  Too late now ;-)

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 9]
```diff
@@ -19,12 +19,18 @@
 
 #define AP_ACCELCAL_POSITION_REQUEST_INTERVAL_MS 1000
 
+#if defined(HAL_BUILD_AP_PERIPH) && !defined(STM32F1)
```
Comment (tridge) [inline libraries/AP_AccelCal/AP_AccelCal.cpp:22]: shouldn't compile accelcal without a full GCS

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 10]
```diff
@@ -110,7 +116,9 @@ void AP_AccelCal::update()
                 uint32_t now = AP_HAL::millis();
                 if (now - _last_position_request_ms > AP_ACCELCAL_POSITION_REQUEST_INTERVAL_MS) {
                     _last_position_request_ms = now;
+#ifndef HAL_NO_GCS
```
Comment (tridge) [inline libraries/AP_AccelCal/AP_AccelCal.cpp:119]: create a define for HAL_INS_WITH_ACCELCAL to go around the whole file

File: libraries/AP_AccelCal/AP_AccelCal.h (hunk 8 above)
Comment (tridge) [reply libraries/AP_AccelCal/AP_AccelCal.h:43]: better HAL_INS_WITH_ACCELCAL

File: libraries/AP_Common/AP_Common.h [hunk 11]
```diff
@@ -162,3 +162,51 @@ template <typename T> void BIT_CLEAR (T& value, uint8_t bitnumber) noexcept {
      static_assert(std::is_integral<T>::value, "Integral required.");
      ((value) &= ~((T)(1U) << (bitnumber)));
  }
+
+
+// helper macro to mark all variables in a varargs list to be marked as unused:
+// based on https://stackoverflow.com/questions/1872220/is-it-possible-to-iterate-over-arguments-in-variadic-macros
```
Comment (tridge) [inline libraries/AP_Common/AP_Common.h:168]: don't need this

File: libraries/GCS_MAVLink/GCS.h [hunk 12]
```diff
@@ -1089,7 +1089,7 @@ void can_printf(const char *fmt, ...);
 
 #else // HAL_NO_GCS
 // empty send text when we have no GCS
-#define GCS_SEND_TEXT(severity, format, args...)
+#define GCS_SEND_TEXT(severity, format, args...) while (0) { (void)severity, (void)format, UNUSED_VARARGS(args) }
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS.h:1092]: remove macro

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by bugobliterator** [comment]
> This does look like it should've been two PRs, one for the `HAL_CANMANAGER_ENABLED` change. Too late now ;-)

it is 😄 , just rebased on top of that PR https://github.com/ArduPilot/ardupilot/pull/17817

**Review by tridge — APPROVED** [review]

**Pull request merged**